webapp/src/Service/EventLogService.php

```diff
@@ -811,33 +853,35 @@ protected function hasAllDependentObjectEvents(Contest $contest, string $type, a
     }
 
     /**
-     * Get the existing event with the same API data as this event
-     * @param Event $event
-     * @return Event|null
+     * Get the existing events with the same API data as these events,
+     * indexed by event endpoint ID. Note that this function expects at least
+     * one event and all events should have the same contest and endpoint.
+     *
+     * @param Event[] $events
+     * @return Event[]
      * @throws NonUniqueResultException
      */
-    protected function getExistingEvent(Event $event)
+    protected function getExistingEvents(array $events)
     {
-        /** @var Event $existingEvent */
-        $existingEvent = $this->em->createQueryBuilder()
-            ->from(Event::class, 'e')
+        $endpointIds = array_map(function(Event $event) {
+            return $event->getEndpointid();
+        }, $events);
+        return $this->em->createQueryBuilder()
+            ->from(Event::class, 'e', 'e.endpointid')
+            ->leftJoin(Event::class, 'e2', Join::WITH,
+                'e2.contest = e.contest AND e2.endpointtype = e.endpointtype AND e2.endpointid = e.endpointid AND e2.eventid > e.eventid'
+            )
             ->select('e')
             ->andWhere('e.contest = :contest')
             ->andWhere('e.endpointtype = :endpoint')
-            ->andWhere('e.endpointid = :endpointid')
-            ->setParameter(':contest', $event->getContest())
-            ->setParameter(':endpoint', $event->getEndpointtype())
-            ->setParameter(':endpointid', $event->getEndpointid())
-            ->setMaxResults(1)
+            ->andWhere('e.endpointid IN (:endpointids)')
+            ->andWhere('e2.eventid IS NULL')
```

Comment:
Why is this there? And I'm wondering how this code does the same thing as the original single-event one when the check for DELETE is missing.

Reply:
* The e2 is there because we want the last event for every endpointid. If you know a better way then doing a LEFT JOIN with all events after the event and it being NULL, we can do this differently.
* DELETE check: ha you're right, I totally missed the DELETE thing. I think the easiest thing to do is to filter from the array all events with ->getAction() === self::ACTION_DELETE before returning the array.

The DELETE thing never popped up in the CI since we almost never delete stuff.

Reply:
DELETE should be fixed now